docker/registry/internal/base.go

```diff
@@ -86,24 +86,49 @@ func (c *baseClient) APIVersion() APIVersion {
 	return APIVersionV1
 }
 
-func (c *baseClient) WrapTransport() error {
-	transport := c.client.Transport
-	if c.repoDetails.IsPrivate() {
-		if !c.repoDetails.BasicAuthConfig.Empty() {
-			transport = newTokenTransport(
-				transport, c.repoDetails.Username, c.repoDetails.Password, c.repoDetails.Auth, "", false,
-			)
-		}
-		if !c.repoDetails.TokenAuthConfig.Empty() {
-			return errors.New(
-				fmt.Sprintf(
-					`only "username" and "password" or "auth" token authorization is supported for registry %q`,
-					c.repoDetails.ServerAddress,
-				),
-			)
+// TransportWrapper wraps RoundTripper.
+type TransportWrapper func(http.RoundTripper, *docker.ImageRepoDetails) (http.RoundTripper, error)
+
+func transportCommon(transport http.RoundTripper, repoDetails *docker.ImageRepoDetails) (http.RoundTripper, error) {
+	if !repoDetails.TokenAuthConfig.Empty() {
+		return nil, errors.New(
+			fmt.Sprintf(
+				`only "username" and "password" or "auth" token authorization is supported for registry %q`,
```

Comment:
Possibly using set notation might help here
`only {username, password} or {auth-token} authorisation is supported for docker registry %q`